meshtastic/interfaces/ble/discovery.py

```
@@ -0,0 +1,536 @@
+    except RuntimeError:
+        try:
+            asyncio.run(_await_close_result(awaitable_result))
+        except Exception:  # noqa: BLE001 - best effort cleanup path
+            logger.debug(
+                "Failed to await close/disconnect for discarded discovery client of type %s.",
+                type(client).__name__,
+                exc_info=True,
+            )
+    else:
+        try:
+            running_loop.create_task(_await_close_result(awaitable_result))
```

Ruff (0.15.2)

[warning] Store a reference to the return value of `running_loop.create_task`

(RUF006)